libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -441,6 +438,33 @@ impl Greenlight {
         Ok(on_chain_balance)
     }
 
+    async fn pending_onchain_balance(
+        &self,
+        peer_channels: &[cln::ListpeersPeersChannels],
+    ) -> Result<u64> {
+        let pending_onchain_balance = peer_channels.iter().fold(0, |a, b| match b.state() {
+            ChanneldShuttingDown | ClosingdSigexchange | ClosingdComplete | AwaitingUnilateral
+            | FundingSpendSeen => a + b.min_to_us_msat.clone().unwrap_or_default().msat,
+
+            // When we  unilaterally close the channel it will get status as `AwaitingUnilateral`
+            // first, but when the closing transaction is confirmed onchain the funds receive status
+            // as `Onchain`. Though if we closed the channel we'll have to wait for the timelock to
+            // pass before the funds can be spent.
+            Onchain => {
+                if b.closer() == cln::ChannelSide::Local
+                    && b.status[b.status.len() - 1].contains("DELAYED_OUTPUT_TO_US")
+                {
```

Comment:
What do you think about using [last()](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.last) here?
```suggestion
                if b.closer() == cln::ChannelSide::Local
                    && b.status.last().is_some_and(|status| status.contains("DELAYED_OUTPUT_TO_US")
                {
```

Reply:
Thank you that looks neater.